src/test/java/io/knotx/launcher/KnotxCommandTest.java

```diff
@@ -0,0 +1,163 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+import org.junit.jupiter.api.parallel.Resources;
+
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.cli.CLI;
+import io.vertx.core.cli.CommandLine;
+import io.vertx.core.impl.launcher.commands.BareCommand;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.Vertx;
+
+@Execution(ExecutionMode.CONCURRENT)
+public class KnotxCommandTest {
+  private Properties backup;
+
+  @BeforeEach
+  void backup() {
+    backup = new Properties();
+    backup.putAll(System.getProperties());
+  }
+
+  @AfterEach
+  void restore() {
+    System.setProperties(backup);
+  }
+
+  @Test
+  @DisplayName("Should read configuration from given path")
+  public void should_read_configuration_from_file(){
+    //given
+    KnotxCommand underTest = new KnotxCommandFactory().create(create(create("run-knox")));
+    underTest.setConfig("./conf/bootstrap.json");
+
+    //when
+    JsonObject config = underTest.getConfiguration();
+
+    //then
+    Assertions.assertNotNull(config);
+  }
+
+  @Test
+  @DisplayName("Should read default configuration")
+  void should_read_default_configuration(){
+    //given
+    KnotxCommand underTest = new KnotxCommandFactory().create(create(create("run-knox")));
+
+    //when
+    JsonObject config = underTest.getConfiguration();
+
+    //then
+    Assertions.assertNotNull(config);
+  }
+
+  @Test
+  @DisplayName("Should return null configuration for not existing file")
+  @ResourceLock(value = Resources.SYSTEM_PROPERTIES, mode = ResourceAccessMode.READ)
+  void should_return_null_for_not_existing_file(){
+    //given
+    KnotxCommand underTest = new KnotxCommandFactory().create(create(create("run-knox")));
+    underTest.setConfig("./not_existing_file.json");
+
+    //when
+    JsonObject config = underTest.getConfiguration();
+
+    //then
+    Assertions.assertNull(config);
+  }
+
+  @Test
+  @DisplayName("Should return default deployment options")
+  @ResourceLock(value = Resources.SYSTEM_PROPERTIES, mode = ResourceAccessMode.READ)
+  void should_return_default_deployment_options(){
+    //given
+    KnotxCommand underTest = new KnotxCommandFactory().create(create(create("run-knox")));
+    JsonObject config = new JsonObject();
+    //when
+    DeploymentOptions deploymentOptions = underTest.getDeploymentOptions(config);
+
+    //then
+    Assertions.assertEquals(false, deploymentOptions.isHa());
```

Comment:
Or `assertFalse` ;)
You may also make static imports to get rid of `Assertions.`